### PR TITLE
Fix #255 - Add phone number validation to sign form

### LINKED ISSUE #255: Add phone number validation to sign form
**Is your feature request related to a problem? Please describe.**
I see that there is a detection of whether an email is correct but not on the phone number. Adding this detection would be coherent if phone number stays mandatory.

**Describe the solution you'd like**
Same interface as validation of e-mail, but for phone numbers.

**Additional context**
After crawling the web, it seems difficult to just verify all phone numbers for all country, so it must maybe be precised that only some format will be accepted.

**Some regex i found**
`^(?:(?:\+|00)33[\s.-]{0,3}(?:\(0\)[\s.-]{0,3})?|0)[1-9](?:(?:[\s.-]?\d{2}){4}|\d{2}(?:[\s.-]?\d{3}){2})$`
But only for french phone numbers, [source](https://askcodez.com/regex-pour-les-numeros-de-telephone-francais.html)

### PR DESCRIPTION
**Pull request by Benbb96**
Fix #255

I added the [django-phonenumber-field](https://github.com/stefanfoulis/django-phonenumber-field) library in order to validate Signature's `phone` field.

### CODE REVIEW

File: pytition/pytition/settings/base.py [hunk 1]
```diff
@@ -275,3 +276,6 @@
 MAINTENANCE_MODE_IGNORE_ADMIN_SITE = True
 # Following setting prevents error 500, due to issue #76 of django_maintenance_mode
 MAINTENANCE_MODE_STATE_FILE_PATH = os.path.join(BASE_DIR, 'maintenance_mode_state.txt')
+
+# Set default region to France in order to accept phone numbers without specifying prefix (+33)
+PHONENUMBER_DEFAULT_REGION = "FR"
```
Comment (Benbb96) [inline pytition/pytition/settings/base.py:281]: I am unsure if we should keep FR as default. Maybe edit the documentation to explain this setting (https://github.com/stefanfoulis/django-phonenumber-field#phonenumber_default_region)

File: pytition/petition/models.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 from django.db import transaction
 from django.urls import reverse
 from django.utils import timezone
+from phonenumber_field.modelfields import PhoneNumberField
```
Comment (fallen) [inline pytition/petition/models.py:14]: Nitpicking here but I'm trying to keep the django imports in one block, and the other imports other blocks.
Could you put it with the block containing tinymce and colorfield?

File: pytition/pytition/settings/base.py (hunk 1 above)
Comment (fallen) [reply pytition/pytition/settings/base.py:281]: I guess it's OK like this, I can't see a better value for this setting. No perfect value here!

File: pytition/pytition/settings/base.py [hunk 3]
```diff
@@ -275,3 +276,6 @@
 MAINTENANCE_MODE_IGNORE_ADMIN_SITE = True
 # Following setting prevents error 500, due to issue #76 of django_maintenance_mode
 MAINTENANCE_MODE_STATE_FILE_PATH = os.path.join(BASE_DIR, 'maintenance_mode_state.txt')
+
+# Set default region to France in order to accept phone numbers without specifying prefix (+33)
```
Comment (fallen) [inline pytition/pytition/settings/base.py:280]: I would reword it as "Set default region to France with regard to accepting phone numbers without international prefix. The +33 prefix for France is therefore optional, but it might be mandatory to add a prefix for other countries. This does not change anything for translations."
In order to make it crystal clear. Better safe than sorry!

**Review by fallen — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution, it looks really good!
Clean, working fine, and has a test.

Please just modify the few nitpicks and it's OK for merging afterward :)

**Comment by Benbb96** [comment]
Thanks for the review. I've applied requested changes !

**Review by fallen — APPROVED** [review]
Looks good to me now, thanks!
